python/tests/mlx_tests.py

```diff
@@ -16,3 +27,23 @@ def setUp(self):
 
     def tearDown(self):
         mx.set_default_device(self.default)
+
+    def compare_dtype(self, mlx_dtype: mx.Dtype, torch_dtype: torch.dtype):
+        assert torch_dtype in dtype_map, "dtype not supported"
+        assert (
+            dtype_map[torch_dtype] == mlx_dtype
+        ), "dtype mismatch got {mlx_dtype} expected {torch_dtype}"
+
+    def compare_torch(
+        self,
+        shapes: List[Tuple[int]],
+        mlx_func: Callable[..., mx.array],
+        torch_func: Callable[..., torch.Tensor],
+    ):
+        mlx_arrs = [mx.random.normal(shape) for shape in shapes]
+        torch_arrs = [torch.from_numpy(np.copy(x)) for x in mlx_arrs]
+        mx_res = mlx_func(*mlx_arrs)
+        torch_res = torch_func(*torch_arrs)
+        assert tuple(mx_res.shape) == tuple(torch_res.shape)
+        self.compare_dtype(mx_res.dtype, torch_res.dtype)
+        np.testing.assert_allclose(mx_res, torch_res.numpy(), rtol=1, atol=1)
```

Comment:
Please remove this as we prefer not to have a non-optional dependency on torch in the test suite.

Reply:
Also those tolerances are really high... I would imagine they should be closer ..

Reply:
removed torch, the tolerances were high as I missed the `init` parameter of 0.25, way better now